decode/layout.ts

```diff
@@ -116,29 +130,38 @@ function process(node: any[] | number[], tagIndex: number): IDecodedNode {
         }
     }
 
-    output.selector = selector(output.id, path, output.tag, attributes);
+    selector(output.id, output.tag, path, attributes);
+    crawler(output.tag, attributes);
     if (hasAttribute) { output.attributes = attributes; }
     if (value) { output.value = value; }
 
     return output;
 }
 
-function selector(id: number, path: string, tag: string, attributes: IAttributes): string {
+function selector(id: number, tag: string, path: string, attributes: IAttributes): void {
     switch (tag) {
         case "STYLE":
         case "TITLE":
         case "LINK":
         case "META":
         case "*T":
         case "*D":
-            return "";
+            break;
         default:
             let s = path && path.length > 0 ? path + tag : tag;
             if ("id" in attributes) { s = `${tag}#${attributes["id"]}`; }
             if ("class" in attributes) { s += `.${attributes["class"].trim().split(" ").join(".")}`; }
             if (ID_ATTRIBUTE in attributes) { s = `*${attributes[ID_ATTRIBUTE]}`; }
-            selectorMap[id] = s;
-            return s;
+            if (s) { selectors.push({ id, checksum: hash(s), selector: s }); }
+            break;
+    }
+}
+
+function crawler(tag: string, attributes: IAttributes): void {
+    switch (tag) {
+        case "LINK":
```

Comment:
>LINK [](start = 14, length = 4)

You should check the link is to a style sheet using the rel tag (rel=stylesheet) as there are many things that can be linked to.. Also record the media property? 
See: [https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link)

Reply:
Good point, fixed.